### DISCUSSION TITLE
Issues Applying MONAI Transformations Directly to List of DICOM File Paths

### DISCUSSION THREAD

**Discussion by sreejaputhumana** [Project-MONAI/MONAI#7608] [Q&A]
**Description**
I am experiencing difficulties applying MONAI transformations directly to a list of DICOM file paths (filtered_patient_file). Despite using transformations that should handle file paths, such as LoadImage, I encounter errors suggesting a mismatch in expected input types. The issue persists even after upgrading to MONAI version 1.3.0. Here is the error message I receive when trying to apply the transformations to the DICOM file paths:

```
RuntimeError: can not find suitable reader for this file: train/train_data/00053/FLAIR/Image-386.dcm.

The above exception was the direct cause of the following exception:

RuntimeError: applying transform <monai.transforms.io.array.LoadImage object at 0x7f83504c0520>
```

**Environment Details**
Monai Version:  1.3
Torch Version: 1.12.1+cu113
Pydicom Version: 2.4.4
Python 3.9.9
Operating System : Windows 11

Code sample:

```
from monai.transforms import Compose, LoadImage, ScaleIntensityRanged

# 'filtered_patient_file' is a list of DICOM file paths
filtered_patient_file = [
    'train/train_data/00053/FLAIR/Image-386.dcm',
    'train/train_data/00053/FLAIR/Image-387.dcm',
    # Add more file paths as necessary
]

# transformations
transforms = Compose([
    LoadImage(image_only=True),  #  to load the image from the file path
    ScaleIntensityRanged(a_min=-1000, a_max=1000, b_min=0.0, b_max=1.0, clip=True),
  
])

processed_images = []
for file_path in filtered_patient_file:
    processed_image = transforms(file_path)  # Apply the transformations
    `processed_images.append(processed_image)`
```

 
**Expected Behavior**
I expected the LoadImage transformation to load each image from the provided file paths and subsequent transformations to be applied successfully.

**Actual Behavior**
I receive a TypeError, indicating an issue with the input type expected by the transformations, despite the use of LoadImage which should handle file paths.

**Additional Context**
I have verified that the file paths are correct and the files are accessible. The issue seems to be related to how the transformations are applied to the list of file paths.

**Questions**:
1.Are there known issues with using LoadImage or other transformations directly with a list of file paths in MONAI?
2.Is there a specific format or preprocessing step required for the file paths before passing them to the MONAI transformations?
3.Could this issue be related to the setup of my MONAI environment or version compatibility?

Any help/suggestions would be appreciated.

**Comment by Lucas-rbnt** [comment]
Hello, your code seems correct, it do not work for every images or only some of them ?

If you look at the source code (https://docs.monai.io/en/stable/_modules/monai/transforms/io/array.html#LoadImage) this error is raised if the image is None or if the reader is None. 

You say that you've checked the file paths and their accessibility. Are you sure that the files are readable as images, for example are you able to really look at the image using pydicom or itk? If it works with pydicom, perhaps try `LoadImage(image_only=True, reader="pydicomreader")` as ITK is the default

Finally you could look in the logs to get a better idea of the problem. (see the code snippet from the `LoadImage`class below, information about a potential error are logged.
```python
try:
    img = reader.read(filename)
except Exception as e:
    err.append(traceback.format_exc())
    logging.getLogger(self.__class__.__name__).debug(e, exc_info=True)
    logging.getLogger(self.__class__.__name__).info(
                            f"{reader.__class__.__name__}: unable to load {filename}.\n"
                        )
```
In the hope that it may have provided some clues

**Reply by sreejaputhumana** [reply]
Thank you for your response. I have confirmed that the issue was indeed with the dataset I was using. Some of the images were not readable as they were not proper DICOM files. After switching to a different dataset with NIfTi images, the LoadImage transformation is working as expected.

I appreciate the suggestion to verify the readability of the files using pydicom or itk. Testing the images with these libraries was helpful in identifying the issue with the dataset.

It turns out that the error was raised because the image files were not proper DICOM images, which led to the None reader problem as mentioned in the LoadImage source code. After ensuring the files were correct and readable, I no longer encountered the error.

Thank you for your help and the insights provided!

**Comment by KumoLiu** [comment]
Hi @sreejaputhumana 

Glad to hear that you're using MONAI for your work, it would be of great help if you could provide more error infomation.

From your given information, I suspect that the issue might be caused due to the absence of the ITK library. By default, MONAI uses the ITK toolkit to read DICOM files. Please verify whether the ITK is installed in your working Python environment.

Hope it helps.

**Reply by sreejaputhumana** [reply]
Thank you so much. I was able to resolve the issue.
